base/dcp_client.go

```diff
@@ -459,11 +473,11 @@ func (dc *DCPClient) openStream(vbID uint16, maxRetries uint32) (err error) {
 	return fmt.Errorf("openStream failed to complete after %d attempts, last error: %w", openRetryCount, openStreamErr)
 }
 
-func (dc *DCPClient) rollback(vbID uint16) (err error) {
+func (dc *DCPClient) rollback(logCtx context.Context, vbID uint16) (err error) {
```

Comment:
I like adding contexts to functions but go idiom says they should always be `ctx` whether used for logging or not.

```suggestion
func (dc *DCPClient) rollback(ctx context.Context, vbID uint16) (err error) {
```